staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -754,14 +788,32 @@ func (o *DiffOptions) Run() error {
 		return err
 	})
 
-	if o.pruner != nil {
-		prunedObjs, err := o.pruner.pruneAll(o.tracker, o.CmdNamespace != "")
-		if err != nil {
-			klog.Warningf("pruning failed and could not be evaluated err: %v", err)
+	if err != nil {
+		return err
+	}
+
+	if o.Prune {
+		var prunedObjs []runtime.Object
+
+		if o.applySet != nil {
+			applyPruneObjs, err := o.applySet.FindAllObjectsToPrune(context.TODO(), o.DynamicClient, o.tracker.visitedUids)
+
+			if err != nil {
+				klog.Warningf("applyset pruning failed and could not be evaluated: %v", err)
+			}
+
+			for _, p := range applyPruneObjs {
+				prunedObjs = append(prunedObjs, p.Object)
+			}
+		}
+
+		if o.pruner != nil {
```

Comment:
Should this be an `else` instead of an `if` ? Isn't the `pruner` mutually exclusive from the `applySet` ? If so, at the very least this mutual exclusivity should be commented.